### PR TITLE
Issue/1883/dispatcher return values 

### LINKED ISSUE #1883: customer.utils.Dispatcher needs return values
Right now `customer.utils.Dispatcher` has a bunch of methods that return `None` where it really makes sense to have a return value. It would make hooking into or overriding `Dispatcher` a lot easier.

Might make sense to have some tests for `Dispatcher` as well.

### PR DESCRIPTION
**Pull request by sasha0**
Fixes #1883.

### CODE REVIEW

**Comment by solarissmoke** [comment]
This looks good to me, but I think tests are necessary because there is some obscure stuff going on with tuples being passed around in various places.

**Comment by sasha0** [comment]
Done, added tests.

File: src/oscar/apps/customer/utils.py [hunk 1]
```diff
@@ -24,65 +24,76 @@ def __init__(self, logger=None):
...
 
-        # Create order communications event for audit
-        if event_type is not None:
-            CommunicationEvent._default_manager.create(
-                order=order, event_type=event_type)
+        self.create_communication_event(order, event_type, dispatched_messages)
+
+    def dispatch_anonymous_messages(self, email, messages):
+        dispatched_messages = {}
+        if email:
```
Comment (solarissmoke) [inline src/oscar/apps/customer/utils.py]: In what situations can `email` be `None`? Is it possible for `order.guest_email` to be missing?

Comment (sasha0) [inline src/oscar/apps/customer/utils.py]: I suppose only in case of email delivery failure.

Comment (sasha0) [inline src/oscar/apps/customer/utils.py]: I don't think `guest_email` can be `None` - we check set it on the checkout gateway page https://github.com/django-oscar/django-oscar/blob/31ea8dcbcce4ccb6bcbdbb08c3fe9b01103fae2c/src/oscar/apps/checkout/views.py#L80-L82.

**Pull request merged**